api/v1alpha1/stack_type.go

```diff
@@ -4,122 +4,125 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 // +kubebuilder:object:root=true
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-// +kubebuilder:resource:path=tinkerbell,scope=Namespaced,categories=tinkerbell,singular=tinkerbell,shortName=tb
+// +kubebuilder:resource:path=stack,scope=Namespaced,categories=stack,singular=stack,shortName=tb
 // +kubebuilder:storageversion
 
-// Tinkerbell represents the tinkerbell stack that is being deployed in the kubernetes where the operator is deployed.
+// Stack represents the tinkerbell stack that is being deployed in the kubernetes where the operator is deployed.
 // Tinkerbell operator watches for different resources such as deployment, services, serviceAccounts, etc. One of those
-// CRs is Tinkerbell which the operator will install the tink-stack based on its specs. Once the CR is deleted,
+// CRs is Stack which the operator will install the tink-stack based on its specs. Once the CR is deleted,
 // the operator will delete all tinkerbell resources.
-type Tinkerbell struct {
+type Stack struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	// Spec describes the desired tinkerbell stack state.
-	Spec TinkerbellStackSpec `json:"spec,omitempty"`
+	Spec StackSpec `json:"spec"`
 }
 
-// TinkerbellStackSpec specifies details of an Tinkerbell setup.
-type TinkerbellStackSpec struct {
+// StackSpec specifies details of an Tinkerbell setup.
+type StackSpec struct {
 	// Version is the Tinkerbell CRD version.
 	Version string `json:"version"`
```

Comment:
Is this a user option or an internal detail? Can I choose which CRD version to deploy?

Reply:
Got it. How is this version info used?
Comment says "Version is the Tinkerbell CRD version" but that doesn't seem right.